lib/goodies/conduit/plug/dead_letter.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Goodies.Conduit.Plug.DeadLetter do
+  use Conduit.Plug.Builder
+
+  @moduledoc """
+  Publishes messages that were nacked or raised an exception to a
+  dead letter destination and ack them to be removed from the regular queue.
+
+  Based on original `Conduit.Plug.DeadLetter`.
+
+  The following options are required:
+
+    * `broker` - The broker to publish the message with.
+    * `publish_to` - The place to publish the message.
+
+  In order for this to work, you must also define an outgoing publish with the same
+  name as the `publish_to` option in your broker.
+
+      publish :error, to: "my_app.error"
+
+  ## Examples
+
+      plug Goodies.Conduit.Plug.DeadLetter, broker: MyApp.Broker, publish_to: :error
+
+  """
+
+  def init(opts) do
+    # Fail if opts are missing
+    _ = Keyword.fetch!(opts, :publish_to)
+    _ = Keyword.fetch!(opts, :broker)
```

Comment:
```suggestion
    broker = Keyword.fetch!(opts, :broker)
    publish_to = Keyword.fetch!(opts, :publish_to)
```

Reply:
```suggestion
    _broker = Keyword.fetch!(opts, :broker)
    _publish_to = Keyword.fetch!(opts, :publish_to)
```
@mgiacomini já que não são usados pq ficam no init